src/components/editor/InputLookupQA.jsx

```diff
@@ -118,21 +119,43 @@ class InputLookupQA extends Component {
                               //Since we don't want promise.all to fail if
                               //one of the lookups fails, we want a catch statement
                               //at this level which will then return the error
-                              return client
-                                .apis
-                                .SearchQuery
-                                .GET_searchAuthority({
-                                  q: query,
-                                  vocab: authority,
-                                  subauthority: subauthority,
-                                  maxRecords: 8,
-                                  lang: language
-                                })
-                                .catch(function(err) {
-                                  console.error("Error in executing lookup against source", err)
-                                  //return information along with the error in its own object
-                                  return { isError: true, errorObject: err }
-                                })
+                              //Subauthorities require a different API call than authorities so need to check if subauthority is available
+                              //The only difference between this call and the next one is the call to Get_searchSubauthority instead of 
+                              //Get_searchauthority.  If there is a way to somehow pass that in a variable name/dynamically, that would be better
+                              if(subauthority) {
```

Comment:
@hudajkhan Re: your comment, how about:

```javascript
const actionFunction = subauthority === undefined ? client.apis.SearchQuery.GET_searchAuthority : client.apis.SearchQuery.GET_searchSubauthority

return actionFunction({
  q: query,
  vocab: authority,
  subauthority: subauthority,
  maxRecords: maxRecords,
  lang: language
})
// ...
```

Reply:
followed this pattern and now in code (tested and works)

Reply:
I'm going to resolve this conversation.  I'm taking the initiative or something but feel free to unresolve if it is not... resolved.

Reply:
awesome!